osu.Game/Screens/Menu/Button.cs

```diff
@@ -35,6 +35,12 @@ public class Button : BeatSyncedContainer, IStateful<ButtonState>
         private readonly Box boxHoverLayer;
         private readonly SpriteIcon icon;
         private readonly string sampleName;
+
+        /// <summary>
+        /// The menu state for which we are visible for.
+        /// </summary>
+        public ButtonSystemState? VisibleState;
```

Comment:
When would this ever be null? / Why should this be supported?

Given that this component is pretty coupled with the button system, why not pass it in the ctor?

Reply:
This can be changed. Initially I thought the settings/back buttons wouldn't need it, but then they did. Also was wary of ctor getting too long, but if you're okay with that then moving sounds good.

Reply:
Either-or re: menu. Not too bigly fussed about that.

The nullability is the main point of concern.